core/core.php

```diff
@@ -41,72 +40,83 @@ public function get_info_sql($fingerprint)
 	{
 		$sql_array = array(
 			'SELECT' => 'pubkey.item_id, pubkey.user_id, pubkey.public_key, pubkey.fingerprint, pubkey.revoked, user.username',
+
 			'FROM' => array(
 				USERS_TABLE => 'user',
 				$this->table_prefix . 'openra_keys' => 'pubkey'
 			),
+
 			'WHERE' => 'pubkey.fingerprint = "' . $this->db->sql_escape($fingerprint) . '"
 				AND pubkey.user_id = user.user_id'
 		);
+
 		$sql = $this->db->sql_build_query('SELECT', $sql_array);
+
 		return $sql;
 	}
 
 	/**
-	 * Returns the sql SELECT statement to fetch user badge data.
-	 * Accepts a fingerprint.
+	 * Returns the sql SELECT statement to fetch user badge data by a fingerprint.
+	 *
 	 * @param var $fingerprint
 	 * @return string DBal SELECT statement
-	 * @link https://wiki.phpbb.com/Dbal.sql_build_query
 	 */
 	public function get_ubadge_sql_by_key($fingerprint)
 	{
 		$sql_array = array(
 			'SELECT' => 'badge.badge_label, badge.badge_icon_24',
+
 			'FROM' => array(
 				USERS_TABLE => 'user',
 				$this->table_prefix . 'openra_keys' => 'pubkey',
 				$this->table_prefix . 'openra_badges' => 'badge',
 				$this->table_prefix . 'openra_user_badges' => 'ubadge',
 			),
+
 			'WHERE' => 'pubkey.fingerprint = "' . $this->db->sql_escape($fingerprint) . '"
 				AND pubkey.user_id = user.user_id
 				AND ubadge.user_id = user.user_id
 				AND ubadge.badge_id = badge.badge_id',
-			'ORDER_BY' => 'ubadge.badge_order',
+
+			'ORDER_BY' => 'ubadge.badge_order'
 		);
 
 		$sql = $this->db->sql_build_query('SELECT', $sql_array);
+
 		return $sql;
 	}
 
 	/**
 	 * Returns the sql SELECT statement to fetch user badge data.
-	 * Accepts a user id.
 	 *
 	 * @param var $user_id
 	 * @return string DBal SELECT statement
-	 * @link https://wiki.phpbb.com/Dbal.sql_build_query
 	 */
 	public function get_ubadge_sql_by_id($user_id)
 	{
+		settype($user_id, 'int');
```

Comment:
Please stick with what I suggested earlier - the entire motivation for casting the variables inline is to make it clear that all input is being sanitized at the point of the sql definition, without having to carefully check how the variables are defined and used previously.

Reply:
Writing it inline didn't work for me when I tested it the first time (probably a syntax error). Then I misunderstood https://stackoverflow.com/questions/13012195/typecast-function-parameters, and erroneously concluded that I had to use the `settype()` function instead.